modin/data_management/functions/groupby_function.py

```diff
@@ -267,9 +272,19 @@ def caller(
         The same type as `query_compiler`
             QueryCompiler which carries the result of GroupBy aggregation.
         """
-        if groupby_args.get("level", None) is None and (
-            not (isinstance(by, (type(query_compiler))) or hashable(by))
-            or isinstance(by, pandas.Grouper)
+        groupby_kwargs, agg_kwargs, agg_args = map(
+            value_or_default,
+            (groupby_kwargs, agg_kwargs, agg_args),
+            (dict, dict, tuple),
+        )
+
+        if (
+            axis != 0
```

Comment:
defaulting to pandas in case of `axis != 0` instead of raising an `AssertionError` (as it was done previously)